main/src/ca/uwaterloo/flix/language/phase/Parser.scala

```diff
@@ -292,7 +293,7 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
       }
 
       rule {
-        Documentation ~ Annotations ~ Modifiers ~ SP ~ keyword("enum") ~ WS ~ Names.Type ~ TypeParams ~ optional(Types.Tuple) ~ Derivations ~ optWS ~ Body ~ SP ~> ParsedAst.Declaration.Enum
+        Documentation ~ Annotations ~ Modifiers ~ SP ~ keyword("enum") ~ WS ~ Names.Type ~ TypeParams ~ optional(Types.Tuple) ~ ((optWS ~ SP ~ Derivations ~ SP) | (SP ~ push(Seq.empty[Name.QName]) ~ SP)) ~ optWS ~ Body ~ SP ~> ParsedAst.Declaration.Enum
```

Comment:
I wanted to move this into a rule by itself, but I had trouble because of the fact that there is no `Rule3` for returning the two positions and the sequence. There is the `RuleN` but I couldn't get it to accept the mixing of types.

Reply:
We can simply add a `Derivations` rule which returns a `ParsedAst.Derivations` and then we are all good 👍